Formula/rure.rb

```diff
@@ -0,0 +1,31 @@
+class Rure < Formula
+  desc "C API for RUst's REgex engine"
+  homepage "https://github.com/rust-lang/regex/tree/master/regex-capi"
+  url "https://github.com/rust-lang/regex/archive/1.5.4.tar.gz"
+  version "0.2.1"
+  sha256 "a91d5b3e1644a1b298ca4ac8e458d693ae268df7fd3307c6d5d12915b5bc3870"
+  license all_of: [
+    "Unicode-TOU",
+    any_of: ["Apache-2.0", "MIT"],
+  ]
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "build", "--manifest-path", "regex-capi/Cargo.toml", "--release"
```

Comment:
```suggestion
    system "cargo", "install", *std_cargo_args(path: "regex-capi")
```
Maybe something like this might work?

Reply:
The problem is that `cargo install` thinks there's nothing to install if the `Cargo.toml` file doesn't specify any executable binary targets - as far as I can tell this is a Cargo limitation, referenced at rust-lang/cargo#8294 and rust-lang/cargo#8317. Here's what happens if we try to use `cargo install`:

```
==> cargo install regex-capi
Last 15 lines from /Users/roland/Library/Logs/Homebrew/rure/01.cargo:
2022-01-30 15:45:44 +0000

cargo
install
--locked
--root
/usr/local/Cellar/rure/0.2.1
--path
regex-capi
```

Additionally, this library doesn't provide a `Cargo.lock` file, so the `--locked` argument in `*std_cargo_args` won't work.

Reply:
Hmmm. I think we'll have to find a way to add an exception in `brew` if we want to do this, then.

Though, looking at other Rust-based formulae that install C libraries: they seem to make use of `cargo-c`. I take it this isn't an install method supported here?

Reply:
Interesting, I hadn't heard of that. Reading about [cargo-c](https://crates.io/crates/cargo-c) now, it looks like it's meant to sort of auto-generate C libraries and headers for Rust code, based on instructions in a file called `capi.rs`. The regex crate provides its own [C header file](https://github.com/rust-lang/regex/blob/master/regex-capi/include/rure.h) and doesn't use `cargo-c` to build the library, so I don't think that will work here.

On the subject of linting/exceptions: `cargo build` has a `--lib` argument that will specifically build library targets. I think the presence of `--lib` in a `cargo build` command could be a good way to differentiate between `cargo build` commands for libraries (which `cargo install` doesn't know what to do with), and `cargo build` commands that really ought to just be `cargo install` with *std_cargo_args.

If that sounds reasonable to you, I can put together a PR in homebrew/brew to make the "use *std_cargo_args" Rubocop rule allow `cargo build --lib` invocations.

Reply:
Added `--lib` here and put up Homebrew/brew#12819 to allow this kind of invocation.